plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestPartitionDrops.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static io.trino.tpch.TpchTable.NATION;
+
+public class TestPartitionDrops
```

Comment:
Can you please add a test where this config plays with `hive.max-partitions-per-scan`

Reply:
yeah, let me see what I can do

Reply:
I looked into this. For non-transactional tables, `hive.max-partitions-per-scan` always has to be higher than the number of partitions to be deleted. Otherwise, we would see `Deletes must match whole partitions for non-transactional tables` error message. So in this case, `hive.max-concurrent-metastore-drops` only comes into play when it's smaller than the number of partitions to be deleted. Since there's essentially no interplay between these two configs, I don't seen the necessity for additional tests.

Reply:
> So in this case, hive.max-concurrent-metastore-drops only comes into play

you mean max-partition-drops-per-transaction?

> has to be higher than the number of partitions to be deleted

I see, then let's validate this relationship at server startup then.

Reply:
yes, I was referring to `max-partition-drops-per-transaction`. The number of partitions to be deleted depends on the actual query. For example, we have a query "DELETE FROM table WHERE partition <= 3;", this will delete 3 partitions. In this case, `hive.max-partitions-per-scan` has to be higher than 3, otherwise, Trino issues an error. This holds for non-transactional tables.

Since for non-transactional tables, `hive.max-partitions-per-scan` has to be equal to or higher than the number of partitions to be deleted as dictated by the query itself, this config doesn't have much to do with `max-partition-drops-per-transaction`

Reply:
so basically the # of partitions that can be deleted will be actually a min of `hive.max-partitions-per-scan` and the config being introduced here.

It makes sense to not explicitly validate that `hive.max-partitions-per-scan` must be >= this config, as practically people may've set lower limits for much more expensive scan operations (compared to drops), and they shouldn't need to add more knobs.

the fact that `hive.max-partitions-per-scan` would also put an upper bound on what we can delete is more like an artifact of how that feature is implemented, so I'm okay with going ahead with the PR in the current shape.